dask/dataframe/io/csv.py

```diff
@@ -628,7 +628,8 @@ def to_csv(df, filename, name_function=None, compression=None, compute=True,
 
     files = open_files(filename, compression=compression, mode=mode,
                        encoding=encoding, name_function=name_function,
-                       num=df.npartitions, **(storage_options or {}))
+                       num=df.npartitions, newline='',
```

Comment:
You've been using `newline=None` as the default, but this is `newline=''`. Do we want that?

Reply:
It seems to me that `None` is the right default for the general `open_files`/`OpenFile` because that's consistent with the built-in `open` (for Python 3) and `io.TextIOWrapper`, and it ensures no changes in behavior for anyone that calls `open_files` without explicitly passing `newline=`.

When opening a file specifically for writing to csv, on the other hand, the choice of using `newline=''` is consistent with Pandas (from what I can tell from chasing it down the rabbit hole; [see here](https://github.com/pandas-dev/pandas/blob/e8132a2901075b96b0d6e74639d5c341bec115fe/pandas/io/common.py#L370)). It's also the [recommended](https://docs.python.org/3/library/csv.html#csv.writer) way to do it from the standard library `csv` module.

Reply:
Agree with this